tools/distpartitioning/dataset_utils.py

```diff
@@ -501,8 +501,14 @@ def get_dataset(input_dir, graph_name, rank, world_size, num_parts, schema_map):
             constants.GLOBAL_TYPE_EID, constants.ETYPE_ID]:
         edge_datadict[col] = []
 
-    for etype_name, etype_info in edge_data.items():
-        edge_info = etype_info[constants.STR_DATA]
+    # Load the edges in the order defined by STR_EDGE_TYPE. 
+    # This order is also used to assign GLOBAL_EID to edges in the function
+    # 'augment_edge_data'
+    # GLOBAL_EID is used to exchange edge features
+
+    for etype_name, etype_id in etype_name_idmap.items():
```

Comment:
The only thing changed is how etype_names are iterated. Why does changing the way it iterates over etypes matter?

Reply:
Hey Xiang, The issue with the search ctr graph was that the EDGE_TYPE list and the EDGE_DATA dictionary has different order of edge types. In the case of search ctr graph, the data was loaded by the nodes in the order specified by the EDGE_DATA order. where as the EDGE_TYPE order was used to assign EDGE_ID's which was creating issues because the edge_ids  were getting flipped. 

Now with this fix, we always assign EDGE_ID's based on the order mentioned in the EDGE_TYPE list.